### PR TITLE
Sipnet debug

### PR DESCRIPTION
**Pull request by helge22a**
## Description

Fixed how model2netcdf.SIPNET handles LE conversion depending on which version of SIPNET is used. @serbinsh @DongchenZ @mdietze @Qianyuxuan 

## Review Time Estimate

- [ ] Immediately
- [ ] Within one week
- [ ] When possible

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue) 
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] My name is in the list of .zenodo.json
- [ ] I have updated the CHANGELOG.md.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: modules/data.atmosphere/R/half_hour_downscale.R [hunk 1]
```diff
@@ -54,16 +54,53 @@ temporal_downscale_half_hour <- function(input_file, output_file, overwrite = TR
...
     dplyr::mutate(relative_humidity = ifelse(.data$relative_humidity > 1, 1, .data$relative_humidity))
-  
+  #convert airTemp and air Press back to K and Pa for met2model fcns
+  #convert C to K
+  C2K <- function(C){
+    K = C + 273.15
+    return(K)
+  }
+  temp.C <- as.matrix(forecast_noaa_ds$air_temperature)
+  temp.K <- apply(temp.C, 1, C2K)
```
Comment (mdietze) [inline modules/data.atmosphere/R/half_hour_downscale.R:94]: replace C2K function with apply(temp.C, 1, udunits2::ud.convert,"Celsius","K")

File: modules/data.atmosphere/R/half_hour_downscale.R [hunk 2]
```diff
@@ -54,16 +54,53 @@ temporal_downscale_half_hour <- function(input_file, output_file, overwrite = TR
...
+    noaa_data_units <- data.frame(time = noaa_data$time, wind_speed = noaa_data$wind_speed, specific_humidity = noaa_data$specific_humidity)
+    airTemp <- noaa_data$air_temperature
+    airPress <- noaa_data$air_pressure
+    #convert K to C
+    K2C <- function(K){
+      C = K - 273.15
+      return(C)
+    }
+    temp.K <- as.matrix(airTemp)
+    temp.C <- apply(temp.K, 1, K2C)
```
Comment (mdietze) [inline modules/data.atmosphere/R/half_hour_downscale.R:67]: replace K2C to udunits

File: modules/data.atmosphere/R/half_hour_downscale.R [hunk 3]
```diff
@@ -54,16 +54,53 @@ temporal_downscale_half_hour <- function(input_file, output_file, overwrite = TR
...
+    noaa_data_units$air_temperature <- temp.C
+    #convert Pa to mb
+    Pa2mb <- function(P){
+      M <- P/100
+      return(M)
+    }
+    press.P <- as.matrix(airPress)
+    press.mb <- apply(press.P, 1, Pa2mb)
```
Comment (mdietze) [inline modules/data.atmosphere/R/half_hour_downscale.R:75]: replace Pa2mb with udunits

File: modules/data.atmosphere/R/half_hour_downscale.R [hunk 4]
```diff
@@ -54,16 +54,53 @@ temporal_downscale_half_hour <- function(input_file, output_file, overwrite = TR
...
+  temp.C <- as.matrix(forecast_noaa_ds$air_temperature)
+  temp.K <- apply(temp.C, 1, C2K)
+  forecast_noaa_ds$air_temperature <- temp.K
+  #convert mb to Pa
+  mb2Pa <- function(M){
+    P = M*100
+    return(P)
+  }
+  press.mb <- as.matrix(forecast_noaa_ds$air_pressure)
+  press.P <- apply(press.mb, 1, mb2Pa)
```
Comment (mdietze) [inline modules/data.atmosphere/R/half_hour_downscale.R:102]: replace mb2Pa with udunits

File: base/all/DESCRIPTION [hunk 5]
```diff
@@ -61,4 +61,4 @@ Copyright: Authors
 LazyLoad: yes
 LazyData: FALSE
 Encoding: UTF-8
-RoxygenNote: 7.0.2
+RoxygenNote: 7.1.2
```
Comment (mdietze) [inline base/all/DESCRIPTION:64]: built under wrong version of roxygen, need to revert

**Review by mdietze — CHANGES_REQUESTED** [review]

File: base/all/DESCRIPTION (hunk 5 above)
Comment (infotroph) [reply base/all/DESCRIPTION:64]: @mdietze @helge22a 7.1.2 is the correct Roxygen version now and is what I see [in this file on the develop branch](https://github.com/PecanProject/pecan/blob/develop/base/all/DESCRIPTION#L80), so I'm puzzled why this change shows in the diff at all

Comment (mdietze) [reply base/all/DESCRIPTION:64]: Thanks @infotroph. @helge22a let's ignore the warnings and focus on the requested changes and the merge conflicts

**Pull request closed without merging**